File: typescript/src/model/product.ts
import {Entity} from './entity'
import {Unit} from './unit'
import {CurrencyAmount} from './currencyAmount'
import {Check} from '../util/check'

export class Product extends Entity<number>{

    constructor(id: number, public name: string, public price: CurrencyAmount, public unit: Unit) {
        super('Product', id);
        Check.required('id', id);
        Check.required('name', name);
        Check.required('price', price);
        Check.required('unit', unit);
    }

    toString(): string {
        return `${this.name} (id=${this.id}, price=${this.price}/${this.unit})`;
    }
}

File: python/models/product.py
from models.currencyamount import CurrencyAmount
from models.unit import Unit
from util.check import required


class Product:
    def __init__(self, id: int, name: str, price: CurrencyAmount, unit: Unit):
        self.id = required('id', id)
        self.name = required('name', name)
        self.price = required('price', price)
        self.unit = required('unit', unit)

    def __str__(self):
        return f'{self.name} (id={self.id}, price= {self.price}/{self.unit})'
